sign_oca/controllers/main.py

```diff
@@ -104,3 +108,125 @@ def get_sign_oca_sign_access(
+        order = searchbar_sortings[sortby]["order"]
+        # Filtering feature
+        searchbar_filters = self._get_my_sign_requests_searchbar_filters()
+        filterby = kwargs.get("filterby") or "all"
+        domain = searchbar_filters.get(filterby, searchbar_filters["all"])["domain"]
+        domain = expression.AND([domain, self.get_sign_requests_domain(request)])
+        SignRequests = request.env["sign.oca.request.signer"].sudo()
+        pager_values = portal_pager(
+            url="/my/sign_requests",
+            total=SignRequests.search_count(domain),
+            page=page,
+            step=self._items_per_page,
+            url_args={},
+        )
+        sign_requests = SignRequests.search(
+            domain,
+            order=order,
+            limit=self._items_per_page,
+            offset=pager_values["offset"],
+        )
+        values = self._prepare_portal_layout_values()
+        values.update(
+            {
+                "sign_requests": sign_requests.sudo() if sign_page else SignRequests,
+                "page_name": "My Sign Requests",
+                "pager": pager_values,
+                "default_url": "/my/sign_requests",
+                "searchbar_sortings": searchbar_sortings,
+                "searchbar_filters": OrderedDict(sorted(searchbar_filters.items())),
+                "sortby": sortby,
+                "filterby": filterby,
+            }
+        )
+        return values
+
+    def _prepare_home_portal_values(self, counters):
+        values = super()._prepare_home_portal_values(counters)
+        if "sign_oca_count" in counters:
+            domain = self.get_sign_requests_domain(request)
+            SignRequests = request.env["sign.oca.request.signer"].sudo()
+            values["sign_oca_count"] = SignRequests.search_count(domain)
+        return values
+
+    @http.route(
+        ["/my/sign_requests", "/my/sign_requests/page/<int:page>"],
+        type="http",
+        auth="public",
+        website=True,
+    )
+    def sign_requests(self, **kwargs):
+        values = self._prepare_sign_portal_rendering_values(sign_page=True, **kwargs)
+        return request.render(
+            "sign_oca.sign_requests",
+            values,
+        )
+
+    @http.route(
+        ["/my/sign/<int:request_id>/download"], type="http", auth="user", website=True
+    )
+    def portal_download_signed(self, request_id, **kw):
+        sign_request = request.env["sign.oca.request"].sudo().browse(request_id)
+        if not sign_request.exists():
+            return request.not_found()
+        # find the signed document attachment
+        attachment = (
+            request.env["ir.attachment"]
+            .sudo()
+            .search(
+                [
+                    ("res_model", "=", "sign.oca.request"),
+                    ("res_id", "=", sign_request.id),
+                    ("res_field", "=", "data"),
+                ],
+                limit=1,
+            )
+        )
```

Comment:
IMO, I think an `attachment_id` store=False field should be created with this domain in `sign.oca.request.signer `to simplify the code https://github.com/OCA/sign/blob/16.0/sign_oca/models/sign_oca_request.py#L289

Furthermore, I believe that the `portal_download_signed()` method is unnecessary because it could be used for the download /web/content/{attachment_id} url (something that the `web` allows: https://github.com/odoo/odoo/blob/215ad0e1c7d74f919c646e40ed23c0bb12cee034/addons/web/controllers/binary.py#L65).